chromium_src/chrome/installer/mini_installer/mini_installer.cc

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2020 The Brave Authors
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#if defined(BRAVE_CHROMIUM_BUILD) && BUILDFLAG(ENABLE_BRAVE_REFERRALS)
+#define BRAVE_RUN_SETUP                                                      \
+  PathString installer_filename;                                             \
+  wchar_t value[MAX_PATH] = {0, };                                           \
+  const bool result =                                                        \
+      RegKey::ReadSZValue(HKEY_CURRENT_USER,                                 \
+                          L"Software\\BraveSoftware\\Promo",                 \
+                          L"StubInstallerPath", value, _countof(value)) ;    \
+  if (result &&                                                              \
+       installer_filename.assign(value) &&                                   \
+       installer_filename.length() != 0) {                                   \
+    ReferralCodeString referral_code;                                        \
+    if (ParseReferralCode(installer_filename.get(), &referral_code)) {       \
+    if (!cmd_line.append(L" --") ||                                          \
+        !cmd_line.append(L"brave-referral-code") ||                          \
+        !cmd_line.append(L"=\"") || !cmd_line.append(referral_code.get()) || \
+        !cmd_line.append(L"\"")) {                                           \
+          return ProcessExitResult(COMMAND_STRING_OVERFLOW);                 \
```

Comment:
I realize you are just copying from the original code, but exiting because we can't append the referral code seems like a really terrible idea. cc @emerick

Reply:
We copied the technique used by all of the other command-line parameters in `mini_installer.cc`, but we could do something else there.